runtime/runtime.go

```diff
@@ -42,7 +42,7 @@ type Node interface {
 	// implemented at. For example, it could be implemented at a VM level for a bare-metal
 	// provider, at a container level, batch job level, etc. In each case, the node must
 	// serve as a host to run containers that are accessible via SSH.
-	InitNode(ctx context.Context, sshKey []types.SSHKey, nodeTypeID string, region *string) (node types.Node, err error)
+	InitNode(ctx context.Context, sshKey []types.SSHKey, hardwareSpec types.HardwareSpec, region *string) (node types.Node, err error)
```

Comment:
[Simple naming is generally better](https://github.com/golang/go/wiki/CodeReviewComments#variable-names). `hardwareSpec -> spec`